From 7e22206d603e5e9fcc56656efead971444f2d3c2 Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Thu, 05 Jun 2025 13:10:14 +0300
Subject: [PATCH] [SW-228856] fit: build fit_services

Context:
We encountered several code duplications related to
fit_foreach_property.
So, we want to create a new fit services infrastructure
to solve this issue in a clean way.
The main Lego pieces are the newly created files:
fit_services.c/h

Current Commit:
In this current commit we build in PPboot the newly
created fit_services.c

General remark:
This infrastructure is intended to support ANY of the
fit related functions that we find today in fit_utils.c
or fit.c
Therefore, from now, please, anyone doing a change related
to the fit files should carefully double-check:
Any unified code can be (and should be) located in
fit_services.c/h
For this, please, follow the example of how it has been
done with fit_foreach_property.

Change-Id: Id058e91f2108e9b2b2f0ace0707d89cd62668cab
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/Makefile b/Makefile
index e24b025..5d77236 100644
--- a/Makefile
+++ b/Makefile
@@ -50,6 +50,7 @@
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/lib_common.c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/boot_desc.c)
 PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/pci.c)
+PPBOOT_SRCS_C += $(shell ls $(EMB_COMMON_MOD_DIR)/lib/common/fit_services.c)
 PPBOOT_APP_S := $(shell ls app/ppboot/crt0_ppboot.S)
 PPBOOT_APP_S += $(shell ls $(HL_DEV_ARCH)/*.S | grep -v lds)
 
@@ -92,7 +93,7 @@
 	@date +'#define PPBOOT_TIME "%T"' >> $@
 
 buildversion-habanalabs: include/baremetal.h $(AUTO_GENERATED_FILE)
-	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV) -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
+	echo "VERSION_STR" | gcc -include $< -I./ -I ./lib/common/ -I ./$(HL_DEV_ARCH)/ -I$(EMB_COMMON_MOD_DIR)/include -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV) -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
 
 $(PCI_PHY_FW_TARGET): $(HL_DEV_ARCH)/pcie_phy_fw.i
 	@cut -d' ' -f2- $< | xxd -r -p | dd status=none bs=16 count=$$(($(PHY_FW_SIZE) / 16)) conv=sync,swab of=$@
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 964786d..b372c31 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -92,6 +92,7 @@
 	$(EMB_COMMON_MOD_DIR)/lib/common/lib_common.o \
 	$(EMB_COMMON_MOD_DIR)/lib/common/boot_desc.o \
 	$(EMB_COMMON_MOD_DIR)/lib/common/pci.o \
+	$(EMB_COMMON_MOD_DIR)/lib/common/fit_services.o \
 	$(EMB_COMMON_MOD_DIR)/lib/preboot/fit_regions.o \
 	$(EMB_COMMON_MOD_DIR)/lib/preboot/fit_rsa.o \
 	$(EMB_COMMON_MOD_DIR)/lib/preboot/fit.o \
